deltachat-ffi/src/lib.rs

```diff
@@ -1389,6 +1389,20 @@ pub unsafe extern "C" fn dc_get_msg_info(
     block_on(message::get_msg_info(&ctx, MsgId::new(msg_id))).strdup()
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn dc_get_msg_html(
+    context: *mut dc_context_t,
+    msg_id: u32,
+) -> *mut libc::c_char {
+    if context.is_null() {
+        eprintln!("ignoring careless call to dc_get_msg_html()");
+        return "".strdup();
```

Comment:
the idea behind returning an empty string instead of NULL was to avoid NULL-pointer-crashes, there was some point somewhere in the past, where things were far less stable, and excluding ffi for being responsible for null-pointer crashes made some sense. but i agree, i have not seen that at many other places as well :)